pkg/solana/client/cache.go

```diff
@@ -0,0 +1,143 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/gagliardetto/solana-go"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/config"
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/monitor"
+)
+
+type CacheGetter[R any] func(ctx context.Context) (res R, slot uint64, err error)
+
+// Cache is a generic implementation for caching data from the chain
+type Cache[R any] struct {
+	services.StateMachine
+
+	// identifier
+	metricName string
+	Account    solana.PublicKey
+	ChainID    string
```

Comment:
It's not actually about these changes, but I’m curious about the choice of using a string for ChainID instead of an enum. I assume it’s for deserialization or improved visibility across different components. I would love to hear your thoughts on this!

Reply:
good question! it could be an enum but having something hard coded would require changes in the case a new solana-based network pops up

string is more useful than an int too because solana does not have chain ids like EVM networks so we use the genesis hash to determine which network we are running on

Reply:
It makes total sense, thanks!